src/search/sql_parser.h

```diff
@@ -41,7 +41,12 @@ struct NumericAtomExpr : WSPad<sor<NumberOrParam, Identifier>> {};
 struct NumericCompareOp : sor<string<'!', '='>, string<'<', '='>, string<'>', '='>, one<'=', '<', '>'>> {};
 struct NumericCompareExpr : seq<NumericAtomExpr, NumericCompareOp, NumericAtomExpr> {};
 
-struct BooleanAtomExpr : sor<HasTagExpr, NumericCompareExpr, WSPad<Boolean>> {};
+struct VectorCompareOp : sor<string<'<', '-', '>'>> {};
```

Comment:
seems `sor` here is useless?

Reply:
This was initially designed to accommodate future operators like `<=>` for cosine distance metrics (see [pgvector](https://github.com/pgvector/pgvector?tab=readme-ov-file#querying)). But upon reconsideration, I think it's probably unnecessary since distance metrics are typically specified during index creation, which doesn't allow users to alter them during queries. 

Thanks for pointing this out; I'll remove the redundant code. <3